internal/util/http.go

```diff
@@ -35,10 +40,23 @@ type HttpClient struct {
 
 // NewHttpClient creates a new client with the given logger.
 func NewHttpClient(logger *log.Logger) HttpClient {
+	return NewHttpClientWithTimeouts(logger, nil, nil)
+}
+
+func NewHttpClientWithTimeouts(logger *log.Logger, responseHeader, total *int) HttpClient {
```

Comment:
Can we use the Timeouts struct here to make the call site a bit more obvious? Then we can create two local variables to hold the timeouts and don't have to do the unchecked deref below.